src/System.Security.Cryptography.Csp/src/System/Security/Cryptography/SafeCryptoHandles.cs

```diff
@@ -182,9 +178,34 @@ internal bool PublicOnly
             }
         }
 
+        internal void SetParent(SafeProvHandle parent)
+        {
+            if (IsInvalid || IsClosed)
+            {
+                return;
+            }
+
+            Debug.Assert(_parent == null);
+            Debug.Assert(!parent.IsClosed);
+            Debug.Assert(!parent.IsInvalid);
+
+            _parent = parent;
+
+            bool ignored = false;
+            parent.DangerousAddRef(ref ignored);
```

Comment:
This probably isn't possible, but just in case, you might want to make this _parent instead of parent.  Otherwise, technically I believe if 'this' were to cease to be rooted elsewhere, it could technically be collected and finalized after setting _parent... then the finalizer would see a non-null parent and call DangerousRelease on it before we added the corresponding ref.  Far fetched, but...